x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -139,9 +187,9 @@ private void executeClusterStateSteps(Index index, String policy, Step step) {
      *            the index settings to extract the {@link StepKey} from.
      */
     public static StepKey getCurrentStepKey(Settings indexSettings) {
-        String currentPhase = LifecycleSettings.LIFECYCLE_PHASE_SETTING.get(indexSettings);
-        String currentAction = LifecycleSettings.LIFECYCLE_ACTION_SETTING.get(indexSettings);
-        String currentStep = LifecycleSettings.LIFECYCLE_STEP_SETTING.get(indexSettings);
+        String currentPhase = LifecycleSettings.LIFECYCLE_NEXT_PHASE_SETTING.get(indexSettings);
```

Comment:
It technically is a "previous" setting from the viewpoint of the internal code, but from an external view (a user looking at the explain or get-settings output) it's the current step, because the age has not yet advanced enough for it